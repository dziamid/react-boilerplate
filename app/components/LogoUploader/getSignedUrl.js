import aws from 'aws-sdk';
import config from 'config';

aws.config.update({
  accessKeyId: config.accessKeyId,
  secretAccessKey: config.secretAccessKey,
});

export default {
  sign(filename, filetype) {
    return new Promise((resolve, reject) => {
      const s3 = new aws.S3();
      const params = {
        Bucket: config.bucketName,
        Key: filename,
        Expires: 60,
        ContentType: filetype,
      };

      s3.getSignedUrl('putObject', params, (err, data) => {
        if (err) {
          reject(err);
        } else {
          resolve(data);
        }
      });
    });
  },
};
