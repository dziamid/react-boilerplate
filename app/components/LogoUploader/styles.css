* {
  box-sizing: border-box;
  padding: 0;
}

body {
  font-family: 'Helvetica Neue';
}

.label_text {
  margin: 0 0 10px;
  font-size: 13px;
  color: #757575;
}

.label_green {
  color: #CBCBCB;
}

.drop {
  display: flex;
}

.upload_info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  color: #A0A0A0;
  padding-left: 20px;
}

.upload_legend {
  margin: 0;
  font-size: 16px;
  font-weight: 300;
  line-height: 1.5;
  color: #757575;
}

.upload_legend_green {
  font-weight: 600;
  color: #1AC5C0;
}

.upload_legend_p {
  margin: -4px 0 0 0;
  font-size: 13px;
  color: #C1C1C1;
  font-weight: 300;
  line-height: 1.9;
}

.logo_container {
  display: inline-block;
  position: relative;
  min-width: 114px;
  border: 1px solid #D3D3D3;
}

.logo {
  max-height: 114px;
  padding: 12px 35px 12px 15px;
}

.logo_img {
  height: 90px;
}

.wrapper {
  margin: 30px 0 30px 5px;
}
