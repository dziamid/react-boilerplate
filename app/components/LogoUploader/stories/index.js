import React from 'react';
import { storiesOf } from '@kadira/storybook';
import LogoUploader from '../index';

storiesOf('LogoUploader', module)
  .add('single logo', () => (
    <LogoUploader />
  ));
