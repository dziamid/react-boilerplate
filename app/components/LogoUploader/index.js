import React from 'react';
import Dropzone from 'react-dropzone';
import Axios from 'axios';
import ContentAdd from 'material-ui/svg-icons/content/add';
import ContentClear from 'material-ui/svg-icons/content/clear';
import styles from './styles.css';
import config from 'config';
import AWSUpload from './getSignedUrl';


const dropzoneStyle = {
  position: 'relative',
  height: '114px',
  width: '114px',
  border: '1px dashed #1ac5c0',
  cursor: 'pointer',
};

const addBtnStyle = {
  width: 48,
  height: 48,
  color: '#1ac5c0',
  position: 'absolute',
  left: '33px',
  top: '33px',
};

const deleteBtnStyle = {
  width: 15,
  height: 15,
  position: 'absolute',
  right: '9px',
  top: '9px',
  cursor: 'pointer',
  color: '#757575',
};

export default class LogoUploader extends React.Component {

  static defaultProps = {
    input: {
      onChange: () => {}
    }
  };

  constructor(props) {
    super(props);

    this.state = {
      uploadedFiles: [],
    };

    this.handleImageDrop = this.handleImageDrop.bind(this);
  }

  async handleImageDrop(file) {
    this.setState({
      uploadedFiles: this.state.uploadedFiles.concat(file),
    });

    const bucketUrl = `https://${config.bucketName}.s3.amazonaws.com`;
    const uploadedFile = file[0];

    const options = {
      headers: {
        'Content-type': uploadedFile.type,
      }
    };

    const imgUrl = `${bucketUrl}/${uploadedFile.name}`;
    this.props.input.onChange(imgUrl);
    const signedUrl = await AWSUpload.sign(uploadedFile.name, uploadedFile.type);
    await Axios.put(signedUrl, uploadedFile, options);
  }

  onDeleteImg(index) {
    this.setState({
      uploadedFiles: this.state.uploadedFiles.filter((f, i) => i !== index),
    });

    this.props.input.onChange(null);
  }

  render() {

    const displayUploader = this.props.multiple === false && this.state.uploadedFiles.length === 1;

    const displayPreviewZone = this.state.uploadedFiles.length > 0;

    const uploader = (
      <div className={styles.drop}>
        <Dropzone style={dropzoneStyle} multiple={false} accept="image/*" onDrop={this.handleImageDrop}>
          <ContentAdd style={addBtnStyle} />
        </Dropzone>
        <div className={styles.upload_info}>
          <h4 className={styles.upload_legend}> Drop or <span className={styles.upload_legend_green}>Select</span> File to Upload</h4>
          <p className={styles.upload_legend_p}>Only .jpg, .jpeg or .png files allowed, no size limit.</p>
        </div>
      </div>
    );

    const previewZone = (
      <div>
        {this.state.uploadedFiles.map((item, index) =>
          <div key={index} className={styles.logo_container}>
            <ContentClear style={deleteBtnStyle} onClick={() => this.onDeleteImg(index)} />
            <div className={styles.logo}>
              <img className={styles.logo_img} src={item.preview} alt="img" />
            </div>
          </div>
        )}
      </div>
    );

    return (
      <div className={styles.wrapper}>
        <div className={styles.label}>
          <p className={styles.label_text}>Logo <span className={styles.label_grey}>optional</span></p>
        </div>
        {displayUploader ? null : uploader}
        {displayPreviewZone ? previewZone : null}
      </div>
    );
  }
}

LogoUploader.defaultProps = {
  multiple: false,
};

