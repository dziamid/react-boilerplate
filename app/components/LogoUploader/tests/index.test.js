/**
 * Testing our LogoUploader component
 */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LogoUploader from '../index';
import injectTapEvent from 'utils/react-tap-event-plugin';
injectTapEvent();

describe('<LogoUploader />', () => {
  it('should render LogoUploader with icon', () => {
    const wrapper = shallow(
      <LogoUploader />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

