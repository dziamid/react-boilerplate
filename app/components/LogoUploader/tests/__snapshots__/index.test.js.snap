exports[`<LogoUploader /> should render LogoUploader with icon 1`] = `
<div>
  <div>
    <p>
      Logo 
      <span>
        optional
      </span>
    </p>
  </div>
  <div>
    <Dropzone
      accept="image/jpeg,image/png"
      disableClick={false}
      disablePreview={false}
      maxSize={Infinity}
      minSize={0}
      multiple={false}
      onDrop={[Function]}
      style={
        Object {
          "border": "1px dashed #1ac5c0",
          "cursor": "pointer",
          "height": "114px",
          "position": "relative",
          "width": "114px",
        }
      }>
      <ContentAdd
        style={
          Object {
            "color": "#1ac5c0",
            "height": 48,
            "left": "33px",
            "position": "absolute",
            "top": "33px",
            "width": 48,
          }
        } />
    </Dropzone>
    <div>
      <h4>
         Drop or 
        <span>
          Select
        </span>
         File to Upload
      </h4>
      <p>
        Only .jpg, .jpeg or .png files allowed, no size limit.
      </p>
    </div>
  </div>
</div>
`;
